ntp-proto/src/filter.rs

```diff
@@ -380,7 +380,10 @@ fn filter_survivor<'a>(
     low: NtpDuration,
     high: NtpDuration,
 ) -> Option<SurvivorTuple<'a>> {
-    if candidate.edge < low || candidate.edge > high {
+    if candidate.edge < low
+        || candidate.edge > high
+        || candidate.endpoint_type != EndpointType::Middle
```

Comment:
Given how interval selection seems to be supposed to function (considering midpoints outside the interval to be falsetickers in it's counting) and that in the further processing things make no real sense if you have duplicate survivors, this seems to be to me the correct way of ensuring at most one SurvivorTuple per peer.

Keeping just any peer whose interval intersects the chosen one seems problematic to me, as it would allow a garbage peer to still enter the calculations if it just advertises a very large root dispersion or root delay.

I will add a comment stating the basics, but agree with you the spec is deeply unclear on this.

Reply:
I disagree with the suggestion to extract this into a method. It is not reused logic and this makes it very clear what we are doing.

Reply:
yeah I did not take the full context into account here. The benefit of extraction would be a name (not reuse, clearly) but doing it inline seems fine given the context.